### PR TITLE
Add basic moderation

### LINKED ISSUE #113: add is_approved flag to Entry
so we can have entries submitted without them automatically showing up, allowing for staff moderation

### PR DESCRIPTION
**Pull request by alanmoo**
Anyone can log in and submit an entry, only staff entries are approved by default.

Admins can approve entries

Little bit of work that will be redundant when we move this into network-api, but at least we don't have a time crunch for that anymore once this gets merged.

Closes #113, See https://github.com/mozilla/network-pulse/issues/501

The implementation of this is a little different, namely we're auto-approving based on email and not "is_staff". We could do a migration so all current users are made staff automatically, but I feel like this approach is better for now because the permission system is going to change a bit once this is wrapped into network-api. The other difference is that approval is a boolean, not some sort of "needs action" indicator. Perhaps as a follow up we could enable an easier approval method than going into each entry to check its status, but this feels like MVP from a network member standpoint.

### CODE REVIEW

**Comment by Pomax** [comment]
I'm seeing several failures when running `manage test` O_o
```
(venv) c:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api>manage test
Creating test database for alias 'default'...
...FFFF.F..F...............
======================================================================
FAIL: test_entries_issue (pulseapi.entries.test_views.TestEntryView)
test filtering entires by issue
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api\pulseapi\entries\test_views.py", line 176, in test_entries_issue
    self.assertEqual(len(entriesJson['results']), 1)
AssertionError: 0 != 1

======================================================================
FAIL: test_entries_pagination (pulseapi.entries.test_views.TestEntryView)
Make sure pagination works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api\pulseapi\entries\test_views.py", line 154, in test_entries_pagination
    self.assertEqual(len(page1Json['results']), 1)
AssertionError: 0 != 1

======================================================================
FAIL: test_entries_search (pulseapi.entries.test_views.TestEntryView)
Make sure filtering searches by tag works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api\pulseapi\entries\test_views.py", line 162, in test_entries_search
    self.assertEqual(len(entriesJson['results']), 1)
AssertionError: 0 != 1

======================================================================
FAIL: test_featured_filter (pulseapi.entries.test_views.TestEntryView)
Entry with all content
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api\pulseapi\entries\test_views.py", line 104, in test_featured_filter
    self.assertEqual(len(entriesJson['results']), 1)
AssertionError: 0 != 1

======================================================================
FAIL: test_get_single_entry_data (pulseapi.entries.test_views.TestEntryView)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api\pulseapi\entries\test_views.py", line 17, in test_get_single_entry_data
    self.assertEqual(response.status_code, 200)
AssertionError: 404 != 200

======================================================================
FAIL: test_post_duplicate_title (pulseapi.entries.test_views.TestEntryView)
Make sure multiple entries can have the same title
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\git\collaborations\mozilla\network-pulse-api\pulseapi\entries\test_views.py", line 37, in test_post_duplicate_title
    self.assertEqual(len(entriesJson['results']), 4)
AssertionError: 0 != 4

----------------------------------------------------------------------
Ran 27 tests in 1.143s

FAILED (failures=6)
Destroying test database for alias 'default'...
```

File: pulseapi/entries/views.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def get_queryset(self):
         ids = self.request.query_params.get('ids', None)
         if ids is not None:
             ids = [ int(x) for x in ids.split(',') ]
-            queryset = Entry.objects.filter(pk__in=ids)
+            queryset = Entry.objects.filter(pk__in=ids, is_approved=True)
```
Comment (Pomax) [inline pulseapi/entries/views.py]: wonder if we should make this conditional on staff permissions, but maybe that should be followup

Comment (alanmoo) [inline pulseapi/entries/views.py]: I think that follow up should also have a "approve entry" button for staff.

File: pulseapi/users/views.py [hunk 2]
```diff
@@ -213,19 +213,6 @@ def callback(request):
         name = userinfo['name']
         email = userinfo['email']
 
-        # For now, we only allow mozilla.com, mozilla.org,
```
Comment (Pomax) [inline pulseapi/users/views.py]: let's put this back in, but let's feature flag it:
```
from django.conf import settings

...

if settings.ALLOW_UNIVERSAL_LOGIN is None:
  code we removed goes here
```

**Pull request merged**